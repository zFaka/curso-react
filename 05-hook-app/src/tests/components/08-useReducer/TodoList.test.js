import {shallow} from "enzyme";
import TodoList from "../../../components/08-useReducer/TodoList";
import {demoTodos} from "../../fixtures/demoTodos";

describe('Test TodoList', () => {

  const handleToggle = jest.fn();
  const handleDelete = jest.fn();

  const wrapper = shallow(
    <TodoList
      todos={demoTodos}
      handleDelete={handleDelete}
      handleToggle={handleToggle}
    />);




  test('Should render', () => {
    expect(wrapper)
      .toMatchSnapshot()
  })




  test('Should has a pair of <TodoListItem/>', () => {


    expect(wrapper.find('TodoListItem').length)
      .toBe(demoTodos.length)

    expect(wrapper
      .find('TodoListItem')
      .at(0)
      .prop('handleDelete')
    )
      .toEqual(expect.any(Function));
  })
})
