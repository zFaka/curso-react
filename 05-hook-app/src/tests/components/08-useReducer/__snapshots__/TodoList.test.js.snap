// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TodoList Should render 1`] = `
<Fragment>
  <ul
    className="list-group list-group-flush"
  >
    <TodoListItem
      handleDelete={[MockFunction]}
      handleToggle={[MockFunction]}
      index={0}
      key="1"
      todo={
        Object {
          "desc": "lrn rct",
          "done": false,
          "id": 1,
        }
      }
    />
    <TodoListItem
      handleDelete={[MockFunction]}
      handleToggle={[MockFunction]}
      index={1}
      key="2"
      todo={
        Object {
          "desc": "lrn mng",
          "done": false,
          "id": 2,
        }
      }
    />
  </ul>
</Fragment>
`;
